### PR TITLE
Feat insta feed

### PR DESCRIPTION
**Pull request by mixcas**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -80,6 +80,54 @@ function new_display_post_thumbnail_column($col, $id){
   }
 }
 
+// Instagram Feed
+function get_instagram_feed() {
+  $instagram_handle = IGV_get_option( '_igv_instagram_handle' );
```
Comment (varffs) [inline functions.php]: what happens if this value is not set?

File: functions.php [hunk 2]
```diff
@@ -80,6 +80,54 @@ function new_display_post_thumbnail_column($col, $id){
   }
 }
 
+// Instagram Feed
+function get_instagram_feed() {
+  $instagram_handle = IGV_get_option( '_igv_instagram_handle' ); 
+  
+  $feed = get_transient( 'instagram_feed' );
+  // Transient yis a piece of informtion that may be or not stored in 
+  // ifast memory instead of in the db. This data es expected to expire,
```
Comment (varffs) [inline functions.php]: wild typos in this para :)

File: functions.php [hunk 3]
```diff
@@ -80,6 +80,54 @@ function new_display_post_thumbnail_column($col, $id){
...
+
+    if( !is_wp_error( $response ) ) {
+      $body = json_decode( $response['body'] );
+      $feed = $body->items;
+
+      // Slice first 5 elements
+      $feed = array_slice($feed, 0, 5);
+
+      // Remove unimportant data
+      foreach( $feed as &$feed_item ) {
```
Comment (varffs) [inline functions.php]: what does `&` do here?

File: functions.php [hunk 4]
```diff
@@ -80,6 +80,54 @@ function new_display_post_thumbnail_column($col, $id){
...
+        unset(
```
Comment (varffs) [inline functions.php]: why do you bother to unset stuff?

File: partials/instagram.php [hunk 5]
```diff
@@ -1,12 +1,39 @@
-<?php 
+<section id="instagram-feed" class="row"> 
+
+<?php
 $instagram_handle = IGV_get_option( '_igv_instagram_handle' ); 
-?>
-<section class="row"> 
+if ($instagram_handle) {
```
Comment (varffs) [inline partials/instagram.php:5]: you call IGV get option twice. Why not just call it in the `get_instagram_feed()` function and check is set up there?

File: partials/instagram.php [hunk 6]
```diff
@@ -1,12 +1,39 @@
...
+  $instagram_feed = get_instagram_feed();
+
+  foreach($instagram_feed as $index => $instagram_item) {
+    $likes = $instagram_item->likes->count;
+    $comments = $instagram_item->comments->count;
+    $url = $instagram_item->images->standard_resolution->url;
+  ?>
+
+  <?php if( $index != 0 ) { ?>
+    <div class="col s1">&#8200;</div>
```
Comment (varffs) [inline partials/instagram.php:18]: what is `&#8200;` ?

File: partials/instagram.php [hunk 7]
```diff
@@ -1,12 +1,39 @@
...
+
+  <div class="instagram-item col s4">
+    <a href="<?php echo $instagram_item->link; ?>">
+
+      <header>
+        <?php echo $likes ? $likes . '<span class="genericon genericon-heart"></span>' : ''; ?>
+        <?php echo $comments ? $comments . '<span class="genericon genericon-comment"></span>' : ''; ?>
+      </header>
+
+      <img src="<?php echo $instagram_item->images->standard_resolution->url; ?>" alt="" width="190px" />
```
Comment (varffs) [inline partials/instagram.php]: do you need to specify the width here? why not let `width: auto` deal with it? Also will it be worth at any point doing src set for this image?

File: functions.php (hunk 2 above)
Comment (mixcas) [inline functions.php]: LOL WTF

File: functions.php (hunk 3 above)
Comment (mixcas) [inline functions.php]: makes `$feed_item` be a reference to the current position in the array. without it `$feed_item` is a copy of the current position. its useful if u need to make direct changes to the iterating array (`$feed`)

File: functions.php (hunk 4 above)
Comment (mixcas) [inline functions.php]: woah, i just realized i made this thinking this data was going to travel thru the wire ala meteor…
is definitely not needed

File: partials/instagram.php (hunk 5 above)
Comment (mixcas) [inline partials/instagram.php:5]: i left this here and send the handler as a parameter to the `get_instagram_feed()` function

File: partials/instagram.php [hunk 8]
```diff
@@ -1,12 +1,40 @@
...
+
+  foreach($instagram_feed as $index => $instagram_item) {
+    $likes = $instagram_item->likes->count;
+    $comments = $instagram_item->comments->count;
+    $url = $instagram_item->images->standard_resolution->url;
+    $caption = $instagram_item->caption->text;
+  ?>
+
+  <?php if( $index != 0 ) { ?>
+    <div class="col s1">&#8200;</div>
```
Comment (mixcas) [inline partials/instagram.php:18]: This is a "space" character. I was gonna ask u cuz the grid fails with empty divs (`.col`)

File: functions.php (hunk 4 above)
Comment (varffs) [inline functions.php]: wow

File: partials/instagram.php (hunk 8 above)
Comment (varffs) [inline partials/instagram.php:18]: ah the styles need the fix for that added [ive been using `min-height: 1px` to force width]

Comment (varffs) [inline partials/instagram.php:18]: any why dont you use non breaking space?

File: functions.php [hunk 9]
```diff
@@ -80,6 +80,43 @@ function new_display_post_thumbnail_column($col, $id){
   }
 }
 
+// Instagram Feed
+function get_instagram_feed($instagram_handle) {
+  
+  if( empty($instagram_handle) ) {
+    return new WP_ERROR('no-instragram-handle', 'Missing Instagram handle');
```
Comment (varffs) [inline functions.php:87]: is this Meteor code too ;]

**Pull request merged**